### PR TITLE
Export gauges for block files after the filtering

### PR DESCRIPTION
**Pull request by hczhu-db**
Deployed to Compactor in `dev-aws-us-west-2` in https://github.com/databricks/universe/pull/506610

### CODE REVIEW

File: pkg/block/fetcher.go [hunk 1]
```diff
@@ -98,8 +102,10 @@ const (
 	// Modified label values.
 	replicaRemovedMeta = "replica-label-removed"
 
-	tenantLabel  = "__tenant__"
-	defautTenant = "__unknown__"
+	tenantLabel    = "__tenant__"
+	defautTenant   = "__unknown__"
+	replicaLabel   = "__replica__"
+	defaultReplica = ""
```
Comment (jnyi) [inline pkg/block/fetcher.go:108]: shall we not use empty label value? maybe just `n_a`?

**Review by jnyi — APPROVED** [review]
lgtm

File: pkg/block/fetcher.go (hunk 1 above)
Comment (hczhu-db) [reply pkg/block/fetcher.go:108]: An empty value label will be removed by M3. It's perfect for our use case.
image

File: pkg/block/fetcher.go [hunk 2]
```diff
@@ -584,7 +606,23 @@ func (f *BaseFetcher) fetch(ctx context.Context, metrics *FetcherMetrics, filter
 			return nil, nil, errors.Wrap(err, "filter metas")
 		}
 	}
-	level.Debug(f.logger).Log("msg", "filtered out block meta data", "before", blockMetaBeforeFilters, "after", len(metas))
+	level.Info(f.logger).Log("msg", "filtered out block meta data", "before", blockMetaBeforeFilters, "after", len(metas), "filters", len(filters))
+	// If filters is empty, it's a global fetch for all block file metadata. metrics.Assigned is for the blocks assigned to this instance.
+	// Therefore, it's skipped to update the gauge.
+	if len(filters) > 0 {
```
Comment (hczhu-db) [inline pkg/block/fetcher.go:606]: @jnyi

**Pull request merged**